### ISSUE TITLE
Analyses to run on the wordlists

### ISSUE THREAD

**Issue by LinguList** [Anaphory/simuling#20]
So far, we try some simple and quick analyses on wordlists. This was important to see how well we model what we want to model, and it helped to sort out some problems. Now we should think of a more systematic way to analyse a given wordlist, first, in order to test how well the model simulates what we want to simulate, second, in order to test how well algorithms can uncover the history.

Regarding the first point, the following measures are useful:

* synonymy per word (we expect some concepts to be expressed by more forms on average than others, this may be compared to actual data)
* cognate density (the "diversity" measure, which has now been enhanced and should be included)
* distances between the languages (here, we can measure Jaccard and Swadesh, they should be transparently explained in the documentation)
* polysemy of cognate sets (how many cross-semantic cognates)

Regarding the second point, I can think of:

* NeighborJoining (implementation in LingPy, doesn't assume a molecular clock)
* UPGMA (also in LingPy, does assume a clock)
* Random tree (to control, maybe even a set of random trees)
* Parsimony on the fixed tree, in order to see how well parsimony on binary states may recover the history of character changes
* Bayesian stuff (will take a long time, but maybe not if we work on small trees)

Let's maybe use this thread to collect more things that we think might be useful to measure.

**Comment by Anaphory** [comment]
Given the discussion, we should make the reconstruction methods more explicit:

 - [ ] NeighborJoining according to … distances from absence/presence coding of within-meaning cognate classes
 - [ ] NeighborJoining according to … distances from absence/presence coding of roots (cross-meaning cognate classes)
 - [ ] UPGMA according to … distances from absence/presence coding of within-meaning cognate classes
 - [ ] UPGMA according to … distances from absence/presence coding of roots (cross-meaning cognate classes)
 - [ ] Random tree generated the same way as the usual method (the “multiple ones” should already come from running multiple simulations, like `-t 100`)
 - [ ] Random tree generated according to the methods used in Bayesian phylogenies (because there might be differences there)
 - Bayesian stuff (maybe just calculate tree posterior probabilities):
    - [ ] Yule pure-birth/covarion/random local clock (using `BEASTling` and beast2)

**Comment by LinguList** [comment]
Yep, exactly. I completely forgot about this. We have essentially two kinds of data to be analysed: pure presence absence with meaning-cognates and without. Ideally even multi-state, but maybe only in Sankoff parsimony with a fixed tree, but NJ, UPGMA, and maybe Bayesian stuff.

If we assume that an external method takes the wordlist and spits out a newick tree (at least for NJ and UPGMA), we could separate this further by putting the method for tree-evaluation in an extra function (here, we also have different methods available in lingpy).

**Comment by Anaphory** [comment]
Yes to pipelines!

**Comment by Anaphory** [comment]
We have a good pipeline infrastructure now. Bayesian analysis is currently not a very simple part of it, so that needs to be put back in, but otherwise, various sampling methods, tree algorithms and tree distances are implemented and more can be easily added through more similar scripts.

**Comment by Anaphory** [comment]
- [ ] RaXML (cf. #45)
